types/scratch-vm/index.d.ts

```diff
@@ -0,0 +1,199 @@
+    arguments?: Record<string, ExtensionArgumentMetadata>;
+}
+
+/** All the metadata needed to register an argument for an extension block. */
+interface ExtensionArgumentMetadata {
+    /** The type of the argument (number, string, etc.) */
+    type: ValueOf<Scratch.extensions.ArgumentType>;
+
+    /** The default value of this argument. */
+    defaultValue?: any;
+
+    /** The name of the menu to use for this argument, if any. */
+    menu?: string;
+}
+
+/** All the metadata needed to register an extension drop-down menu. */
+type ExtensionMenuMetadata = ExtensionDynamicMenu | ExtensionMenuItems;
+
+/** The string name of a function which returns menu items. */
+type ExtensionDynamicMenu = string;
+
+/** Items in an extension menu. */
+type ExtensionMenuItems = Array<ExtensionMenuItemSimple | ExtensionMenuItemComplex>;
+
+/** A menu item for which the label and value are identical strings. */
+type ExtensionMenuItemSimple = string;
+
+/** A menu item for which the label and value can differ. */
+interface ExtensionMenuItemComplex {
+    /** The value of the block argument when this menu item is selected. */
+    value: any;
+
+    /** The human-readable label of this menu item in the menu. */
+    text: string;
+}
+
+interface ScratchExtension {
+    /** Returns data about the extension. */
+    getInfo(): ExtensionMetadata;
+}
+
+/** Indicate the scope for a reporter's value. */
+declare enum ReporterScope {
+    /** This reporter's value is global and does not depend on context. */
+    GLOBAL = "global",
+
+    /**
+     * This reporter's value is specific to a particular target/sprite.
+     * Another target may have a different value or may not even have a value.
+     */
+    TARGET = "target"
+}
+
+declare namespace Scratch {
+    namespace extensions {
+        /** Block argument types. */
+        enum ArgumentType {
+            /** Numeric value with angle picker. */
+            ANGLE = "angle",
+
+            /** Boolean value with hexagonal placeholder. */
+            BOOLEAN = "Boolean",
+
+            /** Numeric value with color picker. */
+            COLOR = "color",
+
+            /** Numeric value with text field. */
+            NUMBER = "number",
+
+            /** String value with text field. */
+            STRING = "string",
+
+            /** String value with matrix field. */
+            MATRIX = "matrix",
+
+            /** MIDI note number with note picker (piano) field. */
+            NOTE = "note",
+
+            /** Inline image on block (as part of the label). */
+            IMAGE = "image"
+        }
+
+        /** Types of blocks. */
+        enum BlockType {
+            /** Boolean reporter with hexagonal shape. */
+            BOOLEAN = "Boolean",
+
+            /** A button (not an actual block) for some special action, like making a variable. */
+            BUTTON = "button",
+
+            /** Command block. */
+            COMMAND = "command",
+
+			/**
```

Comment:
<!--
  1 failure:  The root .editorc...
  0 warning: 
  
  
  DangerID: danger-id-Danger;
  File: types/scratch-vm/index.d.ts;
  Line: 162;
-->
- :no_entry_sign: The root .editorconfig style specifies spaces for whitespace. Please use spaces in new or changed types.